tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,419 @@
+4) Perform cluster operation like restarting mon, osd, overall ceph.target
+5) Remove mon node then add it back while IO keeps going
+6) Remove osd node then add it back while IO keeps going
+7) Initiate OSD scrub operation while IO keeps going
+8) check ceph health and data corruption every time while IO keeps going
+
+Pre-requisites :
+- need client node with ceph-common package, conf and keyring files
+- FIO should be installed on the client.
+
+Environment and limitations:
+- The cluster should have 5 nodes + 1 SSD cache node
+- cluster/global-config-file: config/quincy/upi/octo-5-node-env.yaml
+- Should be Bare-metal.
+
+Support
+- Configure cluster with PWL Cache.
+- Only replicated pool supported, No EC pools."""
+
+import re
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.ceph_admin.helper import add_remove_osd
+from ceph.ceph_admin.orch import Orch
+from ceph.ceph_admin.osd import OSD
+from ceph.parallel import parallel
+from ceph.rados import utils
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.monitor_workflows import MonitorWorkflows
+from tests.rbd.rbd_peristent_writeback_cache import (
+    PersistentWriteAheadLog,
+    fio_ready,
+    get_entity_level,
+)
+from tests.rbd.rbd_utils import initial_rbd_config
+from tests.rbd.test_rbd_persistent_writeback_cache_flush import kill_fio
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def verify_cache(pwl, config, cache_client, image_spec):
+    """Method to verify rbd persistant cache test.
+
+    Args:
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        cache_client (Node): The client node for cache testing.
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Writing data to the image using FIO jobs and checking cache status")
+    with parallel() as p:
+        p.spawn(run_fio, **fio_ready(config, cache_client))
+        p.spawn(
+            pwl.check_cache_file_exists,
+            image_spec,
+            config["fio"].get("runtime", 120),
+            **config,
+        )
+
+
+def restart_service_with_cache_test(
+    service_name, rados_obj, pwl, config, cache_client, image_spec
+):
+    """
+    Restart a Ceph service and verify the persistent write cache.
+
+    Args:
+        service_name (str): The name of the Ceph service to restart.
+        rados_obj (RadosOrchestrator): The RadosOrchestrator instance.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        cache_client (Node): The client node for cache testing.
+        image_spec (str): The image specification for testing.
+    """
+    log.info(f"Test to restarting {service_name} service along with pwl cache test")
+    with parallel() as p:
+        p.spawn(rados_obj.restart_daemon_services, service_name)
```

Comment:
This implementation still doesn't catch any return value from the restart_daemon_services and throw exception right? As we discussed.